profiles/tests/test_gql_federation.py

```diff
@@ -4,53 +4,75 @@
 
 from open_city_profile.tests.asserts import assert_match_error_code
 from profiles.schema import ProfileNode
-from profiles.tests.factories import ProfileFactory
+from profiles.tests.factories import AddressFactory, ProfileFactory
 from services.tests.factories import ServiceConnectionFactory
 
+GRAPHQL_SDL_QUERY = """
+    query {
+        _service {
+            sdl
+        }
+    }
+"""
+
 
-def test_profile_node_exposes_key_for_federation_gateway(anon_user_gql_client):
-    query = """
-        query {
-            _service {
-                sdl
+PROFILE_ENTITY_QUERY = """
+    query ($_representations: [_Any!]!) {
+        _entities(representations: $_representations) {
+            ... on ProfileNode {
+                id
+                firstName
             }
         }
-    """
-
-    executed = anon_user_gql_client.execute(query)
-    assert (
-        'type ProfileNode implements Node  @key(fields: "id")'
-        in executed["data"]["_service"]["sdl"]
-    )
+    }
+"""
 
 
-def test_profile_connection_schema_matches_federated_schema(anon_user_gql_client):
-    query = """
-        query {
-            _service {
-                sdl
+ADDRESS_ENTITY_QUERY = """
+    query ($_representations: [_Any!]!) {
+        _entities(representations: $_representations) {
+            ... on ProfileNode {
+                addresses {
+                    edges {
+                        node {
+                            address
+                            postalCode
+                        }
+                    }
+                }
             }
         }
-    """
+    }
+"""
```

Comment:
Is this how you want to query `AddressNodes` as entities **only**, within a `ProfileNode`? If that's the case, then I think `AddressNode` doesn't have to be an entity at all (doesn't need the `@key` decorator).

If that's not the case, then tests should fetch `AddressNodes` directly as entities. I tried that with the tests and the code seems to deliver exactly that. Here is a patch file [address-node-entity.patch.txt](https://github.com/City-of-Helsinki/open-city-profile/files/6992034/address-node-entity.patch.txt) (the `.txt` suffix is only to persuade Github to accept this file) where I modified the tests to do what I mean.

Let's settle this one question first. Then I have a couple more comments.

Reply:
@akikoskinen That was not the case, I just wasn't so fluent with graphql query syntax and didn't know how to query something on AddressNode directly.

I cannot thank you enough for having written the patch, it made my life so much easier 🙏🏼

I edited the git commit history 👍🏼